### PR TITLE
GPU-1578: Test suite refactoring

### PR DESCRIPTION
**Pull request by plyhun**
## Abstract

Fix some test suite parts, that have no respect over cached entities and respectful transaction.

### General

* [x] Added abstract that describes the change
* [x] Added changelog entry to `/CHANGELOG.adoc`
* [x] Ensured that the change is covered by tests
* [x] Ensured that the change is documented in the docs

### On API Changes

* [x] Checked if the changes are breaking or not
* [x] Added GraphQL API if applicable
* [x] Added Elasticsearch mapping if applicable

### CODE REVIEW

File: tests/tests-core/src/main/java/com/gentics/mesh/cache/ActionContextBranchCacheTest.java [hunk 1]
```diff
@@ -38,12 +40,15 @@ private ProjectBranchNameCache cache() {
...
-	public void testBranchCacheResetOnNewLatest() {
-		BranchResponse branch = call(() -> client().createBranch(projectName(), new BranchCreateRequest().setName("new").setLatest(false)));
-		assertNotNull(branch.getUuid());
-		assertEquals(branch.getName(), "new");
-		assertFalse(branch.getLatest());
-		String branchUuid = branch.getUuid();
+	public void testBranchCacheResetOnNewLatest() throws InterruptedException {
+		String branchUuid = UUIDUtil.randomUUID();
+		waitForJobs(() -> {
+			BranchResponse branch = call(() -> client().createBranch(projectName(), branchUuid, new BranchCreateRequest().setName("new").setLatest(false)));
```
Comment (plyhun) [inline tests/tests-core/src/main/java/com/gentics/mesh/cache/ActionContextBranchCacheTest.java:46]: Flawed test - one has to wait for the branch migration. By now we were lucky to pass the race with the migration job, but Hib 6.x is faster...

File: tests/tests-core/src/main/java/com/gentics/mesh/core/graphql/nativefilter/NativeGraphQLEndpointFilterTest.java [hunk 2]
```diff
@@ -13,12 +12,4 @@
 @MeshTestSetting(testSize = TestSize.FULL, startServer = true, optionChanger = MeshCoreOptionChanger.GRAPHQL_FORCE_NATIVE_FILTER)
 public class NativeGraphQLEndpointFilterTest extends GraphQLEndpointFilterTest {
 
-	/**
```
Comment (plyhun) [inline tests/tests-core/src/main/java/com/gentics/mesh/core/graphql/nativefilter/NativeGraphQLEndpointFilterTest.java:16]: Already implemented.

File: tests/tests-core/src/main/java/com/gentics/mesh/core/node/NodeEndpointTest.java [hunk 3]
```diff
@@ -2291,7 +2291,7 @@ public void testCreateNodeAndDoNotSetRolePermissions() {
 	 */
 	@Test
 	public void testCreateNodeAndSetRolePermissions() {
-		RoleReference anonymous = tx(() -> roles().get("anonymous").transformToReference());
+		RoleReference anonymous = tx(() -> anonymousRole().transformToReference());
```
Comment (plyhun) [inline tests/tests-core/src/main/java/com/gentics/mesh/core/node/NodeEndpointTest.java:2294]: Here and below - disregard of used transactions.

File: tests/tests-core/src/main/java/com/gentics/mesh/test/TestDataProvider.java [hunk 4]
```diff
@@ -395,10 +397,17 @@ private void addUserGroupRoleProject(Tx tx) {
 		tx.commit();
 		tx.<CommonTx>unwrap().data().setEventQueueBatch(batch);
 		batch.dispatch();
-		project = projectDao.create(PROJECT_NAME, null, null, null, userInfo.getUser(),
-			getSchemaContainer("folder").getLatestVersion(), batch);
-		project.addLanguage(tx.languageDao().findByLanguageTag(getEnglish()));
-		project.addLanguage(tx.languageDao().findByLanguageTag(getGerman()));
+		project = projectDao.findByName(PROJECT_NAME);
```
Comment (plyhun) [inline tests/tests-core/src/main/java/com/gentics/mesh/test/TestDataProvider.java:400]: Disregard of parallel test runs.

**Review by npomaroli — APPROVED** [review]

**Pull request merged**